test/feedhandler.cpp

```diff
@@ -0,0 +1,223 @@
+
+	REQUIRE_NOTHROW(feedhandler.get_feed(4));
+	CHECK_THROWS_AS(feedhandler.get_feed(5), std::out_of_range);
+	CHECK_THROWS_AS(feedhandler.get_feed(-1), std::out_of_range);
+}
+
+TEST_CASE("Returns correct number using get_feed_count_by_tag()", "[feedhandler]")
+{
+	FeedHandler feedhandler;
+	std::unique_ptr<configcontainer> cfg(new configcontainer());
+	std::unique_ptr<cache> rsscache(new cache(":memory:", cfg.get()));
+	feedhandler.set_feeds(get_five_empty_feeds(rsscache.get()));
+	feedhandler.get_feed(0)->set_tags({"Chicken", "Horse"});
+	feedhandler.get_feed(1)->set_tags({"Horse", "Duck"});
+	feedhandler.get_feed(2)->set_tags({"Duck", "Frog"});
+	feedhandler.get_feed(3)->set_tags({"Duck", "Hawk"});
+
+	REQUIRE(feedhandler.get_feed_count_per_tag("Ice Cream") == 0);
+	REQUIRE(feedhandler.get_feed_count_per_tag("Chicken") == 1);
+	REQUIRE(feedhandler.get_feed_count_per_tag("Horse") == 2);
+	REQUIRE(feedhandler.get_feed_count_per_tag("Duck") == 3);
+}
+
+TEST_CASE("Correctly returns pos of next unread item", "[feedhandler]")
+{
+	FeedHandler feedhandler;
+	std::unique_ptr<configcontainer> cfg(new configcontainer());
+	std::unique_ptr<cache> rsscache(new cache(":memory:", cfg.get()));
+	const auto feeds = get_five_empty_feeds(rsscache.get());
+	int i = 0;
+	for (const auto& feed : feeds) {
+		const auto item = std::make_shared<rss_item>(nullptr);
+		if ((i % 2) == 0)
+			item->set_unread_nowrite(true);
+		else
+			item->set_unread_nowrite(false);
+		feed->add_item(item);
+		i++;
+	}
+	feedhandler.set_feeds(feeds);
+
+	REQUIRE(feedhandler.get_pos_of_next_unread(0) == 2);
+	REQUIRE(feedhandler.get_pos_of_next_unread(2) == 4);
+}
+
+TEST_CASE("Correctly sorts feeds", "[feedhandler]")
+{
+	SECTION("by none asc")
+	{
+	}
+
+	SECTION("by none desc")
+	{
+	}
+
+	SECTION("by firsttag asc")
+	{
+	}
+
+	SECTION("by firsttag desc")
+	{
+	}
+
+	SECTION("by title asc")
+	{
+	}
+
+	SECTION("by title desc")
+	{
+	}
+
+	SECTION("by articlecount asc")
+	{
+	}
+
+	SECTION("by articlecount desc")
+	{
+	}
+
+	SECTION("by unreadarticlecount asc")
+	{
+	}
+
+	SECTION("by unreadarticlecount desc")
+	{
+	}
+
+	SECTION("by lastupdated asc")
+	{
+	}
+
+	SECTION("by lastupdated desc")
+	{
+	}
+}
+
+TEST_CASE("mark_all_feed_items_read() marks all of feed's items as read", "[feedhandler]")
+{
+	FeedHandler feedhandler;
+	std::unique_ptr<configcontainer> cfg(new configcontainer());
+	std::unique_ptr<cache> rsscache(new cache(":memory:", cfg.get()));
+	const auto feeds = get_five_empty_feeds(rsscache.get());
+	const auto feed = feeds.at(0);
+	for (int j = 0; j < 5; ++j) {
+		const auto item = std::make_shared<rss_item>(nullptr);
+		item->set_unread_nowrite(true);
+		feed->add_item(item);
+	}
+	feedhandler.set_feeds(feeds);
+
+	feedhandler.mark_all_feed_items_read(0);
+
+	for (const auto& item : feed->items()) {
+		REQUIRE(item->unread() == false);
+	}
+}
+
+TEST_CASE("reset_feeds_status() resets status of all feeds", "[feedhandler]")
```

Comment:
I suggest changing this to "reset_feeds_status() changes status of all feeds to "to be downloaded"", which avoids ambiguity of "reset" and even hints at (reminds?) what "status" means for feeds.